synapse/rest/client/v2_alpha/_base.py

```diff
@@ -47,3 +47,14 @@ def client_v2_patterns(path_regex, releases=(0,),
         new_prefix = CLIENT_V2_ALPHA_PREFIX.replace("/v2_alpha", "/r%d" % release)
         patterns.append(re.compile("^" + new_prefix + path_regex))
     return patterns
+
+
+def set_timeline_upper_limit(filter_json, filter_timeline_limit):
+    if filter_timeline_limit < 0:
+        return  # no upper limits
+    if 'room' in filter_json \
+            and 'timeline' in filter_json['room'] \
+            and 'limit' in filter_json['room']['timeline']:
```

Comment:
Style nit: We prefer to never use `\` style new line continuations and instead use brackets.

e.g. something like:

```python
if (
    'room' in filter_json
    and 'timeline' in filter_json['room']
    and 'limit' in filter_json['room']['timeline']
):
    ...
```


(This could also be written as:

```python
timeline = filter_json.get('room', {}).get('timeline', {})
if 'limit' in timeline:
    ...
```

but I'm not sure if that's actually better in this case)